Add proto config files for feature and required class.

KevinGitonga opened the pull request:
Fix #4865
  When this PR is merged it should enable developers to easily access ProfileId which will be wrapped wrapped into all Activities Intents.

## Essential Checklist
<!-- Please tick the relevant boxes by putting an "x" in them. -->
- [x] The PR title and explanation each start with "Fix #bugnum: " (If this PR fixes part of an issue, prefix the title with "Fix part of #bugnum: ...".)
- [x] Any changes to [scripts/assets](https://github.com/oppia/oppia-android/tree/develop/scripts/assets) files have their rationale included in the PR explanation.
- [x] The PR follows the [style guide](https://github.com/oppia/oppia-android/wiki/Coding-style-guide).
- [x] The PR does not contain any unnecessary code changes from Android Studio ([reference](https://github.com/oppia/oppia-android/wiki/Guidance-on-submitting-a-PR#undo-unnecessary-changes)).
- [x] The PR is made from a branch that's **not** called "develop" and is up-to-date with "develop".
- [x] The PR is **assigned** to the appropriate reviewers ([reference](https://github.com/oppia/oppia-android/wiki/Guidance-on-submitting-a-PR#clarification-regarding-assignees-and-reviewers-section)).

## For UI-specific PRs only
<!-- Delete these section if this PR does not include UI-related changes. -->
If your PR includes UI-related changes, then:
- Add screenshots for portrait/landscape for both a tablet & phone of the before & after UI changes
- For the screenshots above, include both English and pseudo-localized (RTL) screenshots (see [RTL guide](https://github.com/oppia/oppia-android/wiki/RTL-Guidelines))
- Add a video showing the full UX flow with a screen reader enabled (see [accessibility guide](https://github.com/oppia/oppia-android/wiki/Accessibility-(A11y)-Guide))
- Add a screenshot demonstrating that you ran affected Espresso tests locally & that they're passing

KevinGitonga committed:
Add proto config files for feature and required class.

KevinGitonga committed:
Add Bazel config on profile Build.bazel.

KevinGitonga committed:
Fix lint buildifier issue on profile Build.bazel.

KevinGitonga committed:
Add profileId decorator to more Activities and refactor to use the same.

KevinGitonga committed:
Remove redundant Kdoc exemption.

KevinGitonga commented:
Hi @BenHenning, I have added the core classes and refactored the implementation to to enable provision of ProfileId to multiple activities intents. Kindly check on the implementation and let me know if i am in the right direction on this. Please note that it works as expected with gradle but not via bazel. This is even after adding bazel config for the decorator class similarly to screen name decorator.Thanks.

KevinGitonga committed:
Add dependency on library to App module to resolve dependency issues.

KevinGitonga committed:
Refactor tests to use CurrentUserProfileIdIntentDecorator.kt instead of "putEXtra".

KevinGitonga committed:
Update to use ProfileId from protoExtra.

BenHenning commented on utility/src/main/java/org/oppia/android/util/profile/CurrentUserProfileIdIntentDecorator.kt:
```diff
@@ -0,0 +1,33 @@
+  fun Intent.decorateWithUserProfileId(profileId: ProfileId) {
+    putProtoExtra(
+      PROFILE_ID_INTENT_DECORATOR,
+      CurrentUserProfile.newBuilder().setProfileId(profileId).build()
```
You could just store ProfileId directly, right? I don't think the extra proto is necessary.

KevinGitonga replied:
Thanks @BenHenning this makes sense, I have updated to use profileId directly instead. Have also added tests covering activities Intents utilizing ProfileId parameter.

KevinGitonga replied:
Fixed CI failures issues by adding the necessary missing bazel dependency and also updating tests to utilize the updated pattern.

BenHenning reviewed (commented):
Thanks @KevinGitonga. Overall I think the approach looks reasonable, though I had one suggestion to simplify the decorator a bit.

That being said, what precisely isn't working for Bazel? The CI checks are passing, so it seems like everything is working correctly. Am I missing something?

KevinGitonga committed:
Add Activity tests for this enhancement and also make changes as advised by reviewer.

KevinGitonga committed:
Merge branch 'develop' into profile_id_provision_feature

BenHenning commented on app/src/main/java/org/oppia/android/app/administratorcontrols/AdministratorControlsActivity.kt:
```diff
@@ -114,15 +115,13 @@ class AdministratorControlsActivity :
       val intent = Intent(context, AdministratorControlsActivity::class.java)
-      intent.putExtra(NAVIGATION_PROFILE_ID_ARGUMENT_KEY, profileId)
       intent.decorateWithScreenName(ADMINISTRATOR_CONTROLS_ACTIVITY)
+      intent.decorateWithUserProfileId(
```
Out of curiosity, if all activities have been updated to use this new decorator is it possible for us just to pass ProfileId around instead of the internal profile int? It might actually simplify this PR a bit since it keeps the data flow a lot cleaner (and avoids awkward cases like this where we need to construct a ProfileId).

KevinGitonga replied:
Thanks Ben this makes sense, updating refactor to this pattern.

BenHenning commented on app/src/main/java/org/oppia/android/app/testing/NavigationDrawerTestActivity.kt:
```diff
@@ -76,7 +81,7 @@ class NavigationDrawerTestActivity :
       RecentlyPlayedActivityParams
         .newBuilder()
-        .setProfileId(ProfileId.newBuilder().setInternalId(internalProfileId).build())
+        .setProfileId(ProfileId.newBuilder().setInternalId(profileId.internalId).build())
```
Couldn't we just use ``profileId`` itself, here? The copy doesn't seem necessary.

KevinGitonga replied:
Updated makes sense.

BenHenning commented on app/src/sharedTest/java/org/oppia/android/app/administratorcontrols/AdministratorControlsActivityTest.kt:
```diff
@@ -714,6 +715,14 @@ class AdministratorControlsActivityTest {
   }
 
+  @Test
+  fun testActivity_createIntent_verifyProfileIdInIntent() {
```
Can we also add a regex check similar to the one we added for screen name to ensure all activity tests have this test present?

KevinGitonga replied:
I have already added this check to activities using ProfileId.

BenHenning replied:
Where is the check? I don't see any additions to file_content_validation_checks.textproto.

Also to be clear, there are two checks needed here that should mirror screen name:

https://github.com/oppia/oppia-android/blob/60b8034f926d27a6bf9c771f8f5d76a6ec7c83f1/scripts/assets/file_content_validation_checks.textproto#L382-L403

KevinGitonga replied:
Updated this thanks.

BenHenning commented on app/src/sharedTest/java/org/oppia/android/app/completedstorylist/CompletedStoryListActivityTest.kt:
```diff
@@ -268,9 +278,11 @@ class CompletedStoryListActivityTest {
-      intended(hasExtra(TopicActivity.getProfileIdKey(), internalProfileId))
       intended(hasExtra(TopicActivity.getTopicIdKey(), FRACTIONS_TOPIC_ID))
       intended(hasExtra(TopicActivity.getStoryIdKey(), FRACTIONS_STORY_ID_0))
+      it.onActivity { it1 ->
```
Here & elsewhere in the PR: let's name the variable ('activity' in this case).

KevinGitonga replied:
Updated this.

BenHenning commented on app/src/sharedTest/java/org/oppia/android/app/home/HomeActivityTest.kt:
```diff
@@ -243,6 +244,21 @@ class HomeActivityTest {
+  }
+
+  @Test
+  fun testActivity_createIntent_verifyCurrentUserProfileId() {
```
Isn't this & the test above verifying the exact same thing? Why do we need two versions of it?

KevinGitonga replied:
Yeah updated to remove one of them.

BenHenning commented on app/src/sharedTest/java/org/oppia/android/app/home/HomeActivityTest.kt:
```diff
@@ -935,9 +951,14 @@ class HomeActivityTest {
       intended(hasExtra(TopicActivity.getTopicIdKey(), FRACTIONS_TOPIC_ID))
       intended(hasExtra(TopicActivity.getStoryIdKey(), FRACTIONS_STORY_ID_0))
+      it.onActivity { it1 ->
+        assertThat(
+          it1.intent.extractCurrentUserProfileId()
+            .internalId
+        ).isEqualTo(internalProfileId1)
```
```suggestion
        assertThat(activity.intent.extractCurrentUserProfileId().internalId)
          .isEqualTo(internalProfileId1)
```

This can wrap more cleanly. Ditto everywhere else in the PR.

KevinGitonga replied:
Updated.

BenHenning commented on app/src/sharedTest/java/org/oppia/android/app/testing/NavigationDrawerActivityDebugTest.kt:
```diff
@@ -315,15 +316,24 @@ class NavigationDrawerActivityDebugTest {
-        )
-      )
+      it.onActivity { it1 ->
+        Truth.assertThat(
```
Nit: please directly import assertThat. Ditto below & elsewhere in the PR.

KevinGitonga replied:
Updated.

BenHenning commented on model/src/main/proto/profile.proto:
```diff
@@ -136,3 +136,5 @@ enum AudioLanguage {
   CHINESE_AUDIO_LANGUAGE =  5;
   BRAZILIAN_PORTUGUESE_LANGUAGE = 6;
 }
+
```
Nit: just one EOF newline.

KevinGitonga replied:
Updated.

BenHenning replied:
Per the latest changes, it seems like there are still 3 newlines here.

BenHenning commented on app/src/sharedTest/java/org/oppia/android/app/topic/TopicActivityTest.kt:
```diff
@@ -164,6 +165,20 @@ class TopicActivityTest {
+      context, 1, FRACTIONS_TOPIC_ID, FRACTIONS_STORY_ID_0
+    ).extractCurrentUserProfileId()
+
+    assertThat(profileIdOne.internalId).isEqualTo(1)
+    assertThat(profileIdTwo.internalId).isEqualTo(1)
```
What's the thought behind verifying two IDs that are actually the same here?

KevinGitonga replied:
Splitted this into two tests as they were previously verifying profileId in two different Activities.

BenHenning commented on utility/src/main/java/org/oppia/android/util/profile/CurrentUserProfileIdIntentDecorator.kt:
```diff
@@ -0,0 +1,32 @@
+
+private const val PROFILE_ID_INTENT_DECORATOR = "PROFILE_ID_INTENT_DECORATOR"
+
+/** Decorator for wrapping a user's profile [ProfileId] within its intent. */
```
```suggestion
/** Decorator that allows an activity to wrap a user's [ProfileId] within its intent. */
```

I wasn't sure what "its" was referring to, so suggesting a rephrasing for clarity.

KevinGitonga replied:
Makes sense updated.

BenHenning commented on utility/src/main/java/org/oppia/android/util/profile/CurrentUserProfileIdIntentDecorator.kt:
```diff
@@ -0,0 +1,32 @@
+/** Decorator for wrapping a user's profile [ProfileId] within its intent. */
+object CurrentUserProfileIdIntentDecorator {
+  /**
+   * Packs the intent with a [ProfileId] proto object that sets it as the current
```
```suggestion
   * Packs [this] intent with a [ProfileId] proto object that sets it as the current
```

(Since it's an extension function).

Ditto for extractCurrentUserProfileId below.

KevinGitonga replied:
Noted.

BenHenning commented on utility/src/main/java/org/oppia/android/util/profile/CurrentUserProfileIdIntentDecorator.kt:
```diff
@@ -0,0 +1,32 @@
+   * [extractCurrentUserProfileId] should be used for retrieving the screen name later.
+   */
+  fun Intent.decorateWithUserProfileId(profileId: ProfileId) {
+    putProtoExtra(
+      PROFILE_ID_INTENT_DECORATOR,
+      profileId
+    )
```
Nit: can fit on one line.

KevinGitonga replied:
Noted.

BenHenning commented on utility/src/main/java/org/oppia/android/util/profile/CurrentUserProfileIdIntentDecorator.kt:
```diff
@@ -0,0 +1,32 @@
+   * Packs the intent with a [ProfileId] proto object that sets it as the current
+   * screen.
+   *
+   * [extractCurrentUserProfileId] should be used for retrieving the screen name later.
```
Not screen name--please update.

KevinGitonga replied:
Noted, Thanks.

BenHenning commented on utility/src/main/java/org/oppia/android/util/profile/CurrentUserProfileIdIntentDecorator.kt:
```diff
@@ -0,0 +1,32 @@
+import org.oppia.android.util.extensions.getProtoExtra
+import org.oppia.android.util.extensions.putProtoExtra
+
+private const val PROFILE_ID_INTENT_DECORATOR = "PROFILE_ID_INTENT_DECORATOR"
```
Suggest instead: "CurrentUserProfileIdIntentDecorator.profile_id_key". I think we're moving over toward this type of pattern for better uniqueness, and the screen name decorator isn't following it yet.

KevinGitonga replied:
Noted. Updated the same. This is just for the key itself right?

BenHenning commented on utility/src/main/java/org/oppia/android/util/profile/CurrentUserProfileIdIntentDecorator.kt:
```diff
@@ -0,0 +1,32 @@
+    )
+  }
+
+  /** Returns the [ProfileId] packed in the intent via [decorateWithUserProfileId]. */
```
Or default instance if there is no profile ID bundled in the intent.

KevinGitonga replied:
Noted, makes sense.

BenHenning commented on utility/src/test/java/org/oppia/android/util/profile/CurrentUserProfileIdIntentDecoratorTest.kt:
```diff
@@ -0,0 +1,38 @@
+class CurrentUserProfileIdIntentDecoratorTest {
+
+  @Test
+  fun testDecorator_decorateWithProfileId_returnsIntentWithCorrectProfileId() {
```
Please update the test names to be more explicit about which API method they're testing. I suggest something like:

``testDecorateWithUserProfileId_emptyProto_addsProtoToIntentAsExtra``

(As an example, not as a replacement for this exact test's name).

That being said, we should be testing both public API methods, and probably the decorate method first (since the extract method depends on it). We should also make sure to test edge cases (e.g. default & missing profile IDs) to make sure the methods do what we expect (and continue staying that way).

KevinGitonga replied:
Thanks Ben have refactored this and also added check for missing profile IDs.

BenHenning reviewed (changes requested):
Thanks @KevinGitonga! I think this looks quite solid! I had a bunch of style comments, and a few of substance. PTAL.

KevinGitonga committed:
Refactor to remove internalId usages to ProfileId directly.

KevinGitonga committed:
Merge branch 'oppia:develop' into profile_id_provision_feature

KevinGitonga committed:
Merge remote-tracking branch 'origin/profile_id_provision_feature' into profile_id_provision_feature

KevinGitonga committed:
Remove redundant exemptions and fix some failing tests.

KevinGitonga committed:
Fix some failing tests.

KevinGitonga committed:
Fix some failing tests.

KevinGitonga committed:
Use getDefautInstance.

KevinGitonga committed:
Fix more failing tests.

KevinGitonga committed:
Fix more failing tests.

KevinGitonga committed:
Fix more failing tests.

KevinGitonga committed:
More refactor PR per reviewer's comments.

KevinGitonga committed:
More refactor for feature.

KevinGitonga committed:
Fix some failing tests.

KevinGitonga committed:
Remove redundant Kdoc exemptions..

KevinGitonga committed:
Refactor Fragments to use similar pattern instead of independently declaring bundle extras.

KevinGitonga committed:
Remove redundant kdoc exemptions.

KevinGitonga committed:
Refactor CurrentUserProfileIdIntentDecorator.kt.

KevinGitonga committed:
Fix failing tests after refactor.

KevinGitonga committed:
Merge branch 'develop' into profile_id_provision_feature

KevinGitonga committed:
Fix failing tests after refactor.

KevinGitonga committed:
Add tests covering all fragments using this feature.

KevinGitonga committed:
Fix some failing tests.

KevinGitonga committed:
Fix some failing tests.

KevinGitonga committed:
Fix some failing tests.

KevinGitonga committed:
Switch from allOf to anyOf, since no way to add ProfileId extra.

KevinGitonga commented:
Hi @BenHenning i have updated this PR as per the comments and also to cover also most Fragments. I have also added tests for the same, Please check. Thanks.

BenHenning commented:
Thanks @KevinGitonga. Given that this is a larger PR, I'll need to take a look at this tomorrow.

BenHenning commented on app/src/main/java/org/oppia/android/app/home/HomeActivity.kt:
```diff
@@ -96,7 +97,7 @@ class HomeActivity :
       RecentlyPlayedActivityParams
         .newBuilder()
-        .setProfileId(ProfileId.newBuilder().setInternalId(internalProfileId).build())
+        .setProfileId(ProfileId.newBuilder().setInternalId(profileId.internalId).build())
```
Can just pass in profileId here.

KevinGitonga replied:
Thanks @BenHenning noted and updated, don't know how i missed this.

KevinGitonga replied:
Yeah updated this.

BenHenning commented on app/src/main/java/org/oppia/android/app/home/promotedlist/PromotedStoryViewModel.kt:
```diff
@@ -56,8 +57,9 @@ class PromotedStoryViewModel(
   }
 
   fun clickOnStoryTile() {
+    val profileId = ProfileId.newBuilder().apply { internalId = internalProfileId }.build()
```
Could a ProfileId be passed into this view model, instead, to avoid needing to create one here?

KevinGitonga replied:
Yeah updated, this too.

BenHenning commented on app/src/main/java/org/oppia/android/app/ongoingtopiclist/OngoingTopicItemViewModel.kt:
```diff
@@ -27,7 +28,8 @@ class OngoingTopicItemViewModel(
 
   fun onTopicItemClicked() {
-    routeToTopic(internalProfileId, topic.topicId)
+    val profileId = ProfileId.newBuilder().apply { internalId = internalProfileId }.build()
```
Ditto here--can we avoid creating a new ProfileId?

KevinGitonga replied:
Updated

BenHenning commented on app/src/main/java/org/oppia/android/app/player/audio/AudioFragment.kt:
```diff
@@ -45,9 +48,9 @@ class AudioFragment :
-    val internalProfileId =
-      arguments!!.getInt(AUDIO_FRAGMENT_PROFILE_ID_ARGUMENT_KEY, /* defaultValue= */ -1)
-    return audioFragmentPresenter.handleCreateView(inflater, container, internalProfileId)
+    val profileId =
+      arguments!!.extractCurrentUserProfileId()
```
Can fit on one line I think.

KevinGitonga replied:
Updated.

BenHenning commented on app/src/main/java/org/oppia/android/app/player/exploration/ExplorationActivityPresenter.kt:
```diff
@@ -43,7 +43,7 @@ private const val TAG_UNSAVED_EXPLORATION_DIALOG = "UNSAVED_EXPLORATION_DIALOG"
 private const val TAG_PROGRESS_DATABASE_FULL_DIALOG = "PROGRESS_DATABASE_FULL_DIALOG"
 private const val TAG_EXPLORATION_FRAGMENT = "TAG_EXPLORATION_FRAGMENT"
-private const val TAG_EXPLORATION_MANAGER_FRAGMENT = "TAG_EXPLORATION_MANAGER_FRAGMENT"
+const val TAG_EXPLORATION_MANAGER_FRAGMENT = "TAG_EXPLORATION_MANAGER_FRAGMENT"
```
Why does this need to be public now?

KevinGitonga replied:
Have reverted this missed reverting as was using the same from tests.

BenHenning commented on app/src/main/java/org/oppia/android/app/settings/profile/ProfileRenameFragmentPresenter.kt:
```diff
@@ -101,6 +101,7 @@ class ProfileRenameFragmentPresenter @Inject constructor(
   }
 
   private fun handleAddProfileResult(result: AsyncResult<Any?>, profileId: Int) {
+    val profileId = ProfileId.newBuilder().apply { internalId = profileId }.build()
```
Since internalId is being accessed above, could profileId just be passed in here to avoid creating a new ProfileId?

KevinGitonga replied:
Yeah Ben sure updated.

BenHenning commented on app/src/main/java/org/oppia/android/app/testing/AudioFragmentTestActivity.kt:
```diff
@@ -19,15 +19,15 @@ class AudioFragmentTestActivity : InjectableAppCompatActivity() {
-    val internalProfileId =
-      intent.getIntExtra(AUDIO_FRAGMENT_TEST_PROFILE_ID_ARGUMENT_KEY, /* defaultValue= */ -1)
-    audioFragmentTestActivityController.handleOnCreate(internalProfileId)
+    val profileId =
+      intent.extractCurrentUserProfileId()
```
Can fit on one line.

KevinGitonga replied:
Yeah updated.

BenHenning commented on app/src/main/java/org/oppia/android/app/topic/lessons/TopicLessonsFragmentPresenter.kt:
```diff
@@ -67,12 +68,11 @@ class TopicLessonsFragmentPresenter @Inject constructor(
     isDefaultStoryExpanded: Boolean
   ): View? {
-    this.internalProfileId = internalProfileId
+    this.profileId = activity.intent.extractCurrentUserProfileId()
```
This should be passed in, instead, since fragments shouldn't be depending on their parent activities directly (as that inverts the dependency chain).

KevinGitonga replied:
Noted, updated the same.

BenHenning commented on app/src/sharedTest/java/org/oppia/android/app/administratorcontrols/AdministratorControlsActivityTest.kt:
```diff
@@ -708,12 +711,23 @@ class AdministratorControlsActivityTest {
+
+  private fun getProfileWithSecondId(): ProfileId {
+    return ProfileId.newBuilder().apply { internalId = 1 }.build()
+  }
```
Nit: need newline after this.

KevinGitonga replied:
Added newline.

BenHenning reviewed (changes requested):
Thanks @KevinGitonga. I reviewed up to the start of the tests, but will need to take an additional pass on Monday to finish it up.

KevinGitonga committed:
Merge branch 'oppia:develop' into profile_id_provision_feature

KevinGitonga committed:
Make changes per reviewer's comments also self review viewmodels and update some per the pattern.

KevinGitonga committed:
Fix failing tests.

KevinGitonga committed:
Update some files to new pattern.

KevinGitonga committed:
Add omitted bazel config.

KevinGitonga commented:
Hi @BenHenning i have updated this PR and also made changes after self reviewing it looks good to me. PTAL. Thanks.

BenHenning commented:
Hi @KevinGitonga I couldn't quite get to this today, however I also just noticed there are quite a few conflicts (probably due to the big PRs that were just merged). Please do **not** updated to the latest develop. I'll review the rest of the PR, first (otherwise for PRs this large with that many conflicts I'll probably have a tougher incremental review if it's not just the merge part).

BenHenning commented:
I plan to look at this tomorrow, barring any high priority things that come my way. :)

BenHenning commented on app/src/sharedTest/java/org/oppia/android/app/devoptions/DeveloperOptionsActivityTest.kt:
```diff
@@ -165,14 +168,22 @@ class DeveloperOptionsActivityTest {
+    val profileId = createDeveloperOptionsActivityIntent(profileId = this.profileId)
+      .extractCurrentUserProfileId()
+
+    assertThat(profileId.internalId).isEqualTo(this.profileId.internalId)
```
Here & elsewhere in the PR: I suggest instead importing ``import com.google.common.truth.extensions.proto.LiteProtoTruth.assertThat`` & just comparing the proto directly rather than the internal ID.

KevinGitonga replied:
Thanks, updated.

BenHenning replied:
Are all such cases updated? It seems like some of the test changes are still verifying internalId unless I'm missing something.

BenHenning commented on app/src/sharedTest/java/org/oppia/android/app/home/HomeActivityTest.kt:
```diff
@@ -210,18 +212,17 @@ class HomeActivityTest {
-  private val internalProfileId: Int = 0
-  private val internalProfileId1: Int = 1
-  private val longNameInternalProfileId: Int = 3
+  private lateinit var longNameInternalProfileId: ProfileId
```
Drop 'internal' since this isn't actually an internal ID.

KevinGitonga replied:
Updated.

BenHenning commented on app/src/sharedTest/java/org/oppia/android/app/home/HomeActivityTest.kt:
```diff
@@ -238,14 +239,37 @@ class HomeActivityTest {
 
+  @Test
+  fun testHomeActivity_createIntent_verifyProfileIdInIntent() {
+    val profileId = createHomeActivityIntent(profileId)
+      .extractCurrentUserProfileId()
```
Here & elsewhere in the PR: please keep on one line if it fits, and if not break after the ``=`` (the higher syntactical element).

KevinGitonga replied:
Noted, updated,

BenHenning commented on app/src/sharedTest/java/org/oppia/android/app/home/HomeActivityTest.kt:
```diff
@@ -238,14 +239,37 @@ class HomeActivityTest {
+  }
+
+  @Test
+  fun testHomeActivity_createIntentWithProfileId_verifyProfileIdInBundle() {
```
How is this different than the test above?

KevinGitonga replied:
Bottom one verifies if profileId is in fragment bundle while the other checks the same in intent.

BenHenning commented on app/src/sharedTest/java/org/oppia/android/app/home/promotedlist/PromotedStoryListViewModelTest.kt:
```diff
@@ -132,6 +134,7 @@ class PromotedStoryListViewModelTest {
   @Before
   fun setUp() {
     setUpTestApplicationComponent()
+    profileId = ProfileId.getDefaultInstance()
```
Shouldn't this be creating with an internal ID of 1 (for parity with the test's previous behavior)?

KevinGitonga replied:
yeah updated this for parity.

BenHenning commented on app/src/sharedTest/java/org/oppia/android/app/mydownloads/MyDownloadsActivityTest.kt:
```diff
@@ -126,12 +128,22 @@ class MyDownloadsActivityTest {
     val screenName = MyDownloadsActivity.createMyDownloadsActivityIntent(
       ApplicationProvider.getApplicationContext(),
-      1
+      ProfileId.newBuilder().setInternalId(0).build()
```
Shouldn't the internal ID here be 1 per previous behavior?

KevinGitonga replied:
Yeah updated this, Thanks.

BenHenning commented on app/src/sharedTest/java/org/oppia/android/app/ongoingtopiclist/OngoingTopicListActivityTest.kt:
```diff
@@ -243,16 +272,21 @@ class OngoingTopicListActivityTest {
       intended(hasExtra(TopicActivity.getTopicIdKey(), RATIOS_TOPIC_ID))
+      it.onActivity { activity ->
+        assertThat(
+          activity.intent.extractCurrentUserProfileId()
+            .internalId
```
Nit: should fit on one line (ditto elsewhere in this file & PR).

KevinGitonga replied:
Updated.

BenHenning commented on app/src/sharedTest/java/org/oppia/android/app/settings/profile/ProfileEditActivityTest.kt:
```diff
@@ -132,11 +132,17 @@ class ProfileEditActivityTest {
     Intents.init()
     setUpTestApplicationComponent()
     testCoroutineDispatchers.registerIdlingResource()
+    adminProfileId = ProfileId.newBuilder().apply { internalId = 1 }.build()
+    userProfileId = ProfileId.getDefaultInstance()
```
These are backwards--admin corresponds to ID 0.

KevinGitonga replied:
Updated.

BenHenning commented on app/src/sharedTest/java/org/oppia/android/app/spotlight/SpotlightFragmentTest.kt:
```diff
@@ -338,6 +341,24 @@ class SpotlightFragmentTest {
+        )
+        val profileId = fragment?.arguments?.extractCurrentUserProfileId()
+
+        assertThat(profileId).isEqualTo(ProfileId.getDefaultInstance())
```
Here & elsewhere in the PR: when using ``LiteProtoTruth.assertThat`` you can use ``isEqualToDefaultInstance`` to make this check a bit clearer.

KevinGitonga replied:
updated.

BenHenning replied:
It seems like some test classes that include these kinds of tests still aren't using the LiteProtoTruth's assertThat function--did you add the import to all tests that are checking a ProfileId?

BenHenning commented on app/src/sharedTest/java/org/oppia/android/app/topic/conceptcard/ConceptCardFragmentTest.kt:
```diff
@@ -374,6 +377,23 @@ class ConceptCardFragmentTest {
   }
 
+  @Test
+  fun testConceptCardFragment_createIntentWithProfileId_verifyProfileIdInBundle() {
```
Isn't this verifying that the profile ID is present in fragment arguments, not an intent bundle? Please update, and all other similar test names introduced in this PR.

KevinGitonga replied:
Updated.

BenHenning commented on utility/src/test/java/org/oppia/android/util/profile/CurrentUserProfileIdIntentDecoratorTest.kt:
```diff
@@ -0,0 +1,76 @@
+  @Test
+  fun testDecorator_withoutProfileId_returnsIntentWithDefaultProfileId() {
+    val currentProfileId = Intent().extractCurrentUserProfileId()
+    assertThat(currentProfileId).isEqualTo(ProfileId.getDefaultInstance())
```
Prefer using TruthProtoSubject here & elsewhere in this test suite, too, per the earlier comment.

KevinGitonga replied:
Updated..

BenHenning commented on utility/src/test/java/org/oppia/android/util/profile/CurrentUserProfileIdIntentDecoratorTest.kt:
```diff
@@ -0,0 +1,76 @@
+
+  @Test
+  fun testDecorateBundleWithUserProfileId_emptyProfileId_returnsDefaultProfileIdInstance() {
+    val profileId = ProfileId.newBuilder().apply {}.build()
```
Here & above: you can use ``getDefaultInstance`` for an "empty" proto.

KevinGitonga replied:
Updated.

BenHenning reviewed (changes requested):
Thanks @KevinGitonga! Took a whole pass on the PR.

Please go ahead and update this branch to the latest develop, and then send this back after comments are addressed & CI is passing. I suspect we'll just need a few minor follow-up review passes after this latest review to wrap up the PR. I'm quite happy with how much this is broadly cleaning up profile ID usage, too.

KevinGitonga committed:
Merge branch 'develop' into profile_id_provision_feature

# Conflicts:
#	app/src/main/java/org/oppia/android/app/devoptions/DeveloperOptionsActivity.kt
#	app/src/main/java/org/oppia/android/app/devoptions/markchapterscompleted/MarkChaptersCompletedActivity.kt
#	app/src/main/java/org/oppia/android/app/devoptions/markchapterscompleted/MarkChaptersCompletedActivityPresenter.kt
#	app/src/main/java/org/oppia/android/app/devoptions/markchapterscompleted/MarkChaptersCompletedFragment.kt
#	app/src/main/java/org/oppia/android/app/devoptions/markchapterscompleted/MarkChaptersCompletedFragmentPresenter.kt
#	app/src/main/java/org/oppia/android/app/devoptions/markchapterscompleted/testing/MarkChaptersCompletedTestActivity.kt
#	app/src/main/java/org/oppia/android/app/devoptions/testing/DeveloperOptionsTestActivity.kt
#	app/src/main/java/org/oppia/android/app/settings/profile/ProfileEditFragmentPresenter.kt
#	app/src/main/java/org/oppia/android/app/story/StoryFragmentPresenter.kt
#	app/src/sharedTest/java/org/oppia/android/app/devoptions/MarkChaptersCompletedActivityTest.kt
#	app/src/sharedTest/java/org/oppia/android/app/devoptions/MarkChaptersCompletedFragmentTest.kt
#	app/src/sharedTest/java/org/oppia/android/app/settings/profile/ProfileEditActivityTest.kt
#	app/src/sharedTest/java/org/oppia/android/app/settings/profile/ProfileEditFragmentTest.kt
#	app/src/sharedTest/java/org/oppia/android/app/testing/NavigationDrawerActivityDebugTest.kt
#	app/src/sharedTest/java/org/oppia/android/app/topic/TopicFragmentTest.kt
#	app/src/test/java/org/oppia/android/app/home/HomeActivityLocalTest.kt
#	app/src/test/java/org/oppia/android/app/topic/info/TopicInfoFragmentLocalTest.kt
#	app/src/test/java/org/oppia/android/app/topic/lessons/TopicLessonsFragmentLocalTest.kt
#	app/src/test/java/org/oppia/android/app/topic/revisioncard/RevisionCardActivityLocalTest.kt

KevinGitonga committed:
Revert "Merge branch 'develop' into profile_id_provision_feature"

This reverts commit e295d16d84a845f6a8e4019d4f873d33b569bb19, reversing
changes made to 03601e091f06500ae116f26f461af9018c331bca.

KevinGitonga committed:
Revert "Merge branch 'develop' into profile_id_provision_feature"

This reverts commit e295d16d84a845f6a8e4019d4f873d33b569bb19, reversing
changes made to 03601e091f06500ae116f26f461af9018c331bca.

KevinGitonga committed:
Merge remote-tracking branch 'origin/profile_id_provision_feature' into profile_id_provision_feature

KevinGitonga committed:
Revert "Revert "Merge branch 'develop' into profile_id_provision_feature""

This reverts commit 9092288197e6a9283d2df6ff3ecd3f2c75d5021e.

KevinGitonga committed:
Revert and merge some files to align PR.

KevinGitonga committed:
Fix some failing tests.

KevinGitonga committed:
Fix some failing tests.

KevinGitonga committed:
updates based on reviewer's comments.

KevinGitonga committed:
Fix some failing tests.

KevinGitonga committed:
Merge branch 'develop' into profile_id_provision_feature

# Conflicts:
#	app/src/main/java/org/oppia/android/app/home/recentlyplayed/RecentlyPlayedFragmentPresenter.kt

KevinGitonga committed:
Revert removed file after merge.

KevinGitonga committed:
Refactor RecentlyPlayedFragmentTest.kt file.

KevinGitonga committed:
Fix failing tests by reverting recently_played_fragment.xml file.

KevinGitonga commented:
Hey @BenHenning i have updated this PR and aligned it per your comments. Also it might have picked up some unrelated changes while merging to latest develop which were not conflicting on a few files which i had to roll back to prevent for unrelated changes. PTAL and advice, Thanks.

BenHenning commented on app/src/main/java/org/oppia/android/app/devoptions/markchapterscompleted/ChapterSelector.kt:
File accidentally re-added during merge with develop?

KevinGitonga replied:
Reverted this.

BenHenning commented on app/src/main/java/org/oppia/android/app/home/recentlyplayed/PromotedStoryListAdapter.kt:
File accidentally re-added during merge with develop?

KevinGitonga replied:
Reverted this.

BenHenning commented on app/src/main/java/org/oppia/android/app/home/recentlyplayed/RecentlyPlayedFragmentPresenter.kt:
Extra changes accidentally re-added during merge with develop? Perhaps restart with the current version of the file and then manually re-add the ProfileId changes (since there are presumably not many).

KevinGitonga replied:
Hi Ben this has to pick up changes since latest file on develop has changed to using bindable adapter, couldn't find a way to intergrate this without picking up changes from develop.

BenHenning commented on app/src/main/java/org/oppia/android/app/settings/profile/ProfileEditFragmentPresenter.kt:
Seems like there are extra changes in this file as well--please fix.

KevinGitonga replied:
Hi Ben some changes i could not avoid picking as they either would cause error in my feature or breakage of some tests. Thanks.

BenHenning commented on app/src/main/java/org/oppia/android/app/story/StoryFragmentPresenter.kt:
```diff
@@ -65,13 +65,16 @@ class StoryFragmentPresenter @Inject constructor(
 
-  @Inject lateinit var storyViewModel: StoryViewModel
-  @Inject lateinit var accessibilityService: AccessibilityService
+  @Inject
+  lateinit var storyViewModel: StoryViewModel
+
+  @Inject
+  lateinit var accessibilityService: AccessibilityService
```
Please revert style-only change.

KevinGitonga replied:
Updated.

BenHenning commented on app/src/main/java/org/oppia/android/app/devoptions/markchapterscompleted/MarkChaptersCompletedFragment.kt:
Are there extra changes here that need to be reverted?

KevinGitonga replied:
Hi Ben while also reverting some changes per this https://github.com/oppia/oppia-android/pull/4869#discussion_r1155037875 and this https://github.com/oppia/oppia-android/pull/4869#discussion_r1155037746 i had to intergrate some of the changes from develop to either fix failing tests and fix erros with my PR. Thanks.

BenHenning commented on app/src/main/java/org/oppia/android/app/devoptions/markchapterscompleted/MarkChaptersCompletedFragmentPresenter.kt:
Are there extra changes here that need to be reverted?

KevinGitonga replied:
Hi Ben while also reverting some changes per this https://github.com/oppia/oppia-android/pull/4869#discussion_r1155037875 and this https://github.com/oppia/oppia-android/pull/4869#discussion_r1155037746 i had to intergrate some of the changes from develop to either fix failing tests and fix erros with my PR. Thanks.

BenHenning commented on app/src/main/res/layout/recently_played_fragment.xml:
Are there extra changes here that need to be reverted?

KevinGitonga replied:
Hi Ben while also reverting some changes per this https://github.com/oppia/oppia-android/pull/4869#discussion_r1155037875 and this https://github.com/oppia/oppia-android/pull/4869#discussion_r1155037746 i had to intergrate some of the changes from develop to either fix failing tests and fix erros with my PR. Thanks.

BenHenning commented on app/src/sharedTest/java/org/oppia/android/app/topic/revision/TopicRevisionFragmentTest.kt:
```diff
@@ -278,8 +281,25 @@ class TopicRevisionFragmentTest {
+      profileId,
+      topicId = FRACTIONS_TOPIC_ID
+    ).use {
+      it.onActivity {
+        activity ->
```
```suggestion
      it.onActivity { activity ->
```

Wrapping is a bit off here.

KevinGitonga replied:
Updated.

BenHenning commented on app/src/test/java/org/oppia/android/app/topic/lessons/TopicLessonsFragmentLocalTest.kt:
```diff
@@ -94,21 +95,23 @@ private const val TEST_STORY_ID = "GJ2rLXRKD5hw"
 )
 class TopicLessonsFragmentLocalTest {
-  @get:Rule val initializeDefaultLocaleRule = InitializeDefaultLocaleRule()
+  @get:Rule
+  val initializeDefaultLocaleRule = InitializeDefaultLocaleRule()
```
Please revert style change.

KevinGitonga replied:
Updated.

BenHenning commented on app/src/sharedTest/java/org/oppia/android/app/topic/practice/TopicPracticeFragmentTest.kt:
```diff
@@ -379,14 +383,34 @@ class TopicPracticeFragmentTest {
+      onView(isRoot()).perform(orientationLandscape())
+      scrollToPosition(position = 5)
+
+      it.onActivity {
+        activity ->
```
```suggestion
      it.onActivity { activity ->
```

Wrapping is off here. Ditto everywhere else in the PR.

KevinGitonga replied:
Updated.

BenHenning commented on app/src/sharedTest/java/org/oppia/android/app/topic/lessons/TopicLessonsFragmentTest.kt:
```diff
@@ -687,12 +689,12 @@ class TopicLessonsFragmentTest {
       }.build()
       intended(
-        allOf(
+        anyOf(
```
This change seems incorrect as it changes the meaning of the assertion--why was it made? Ditto above.

KevinGitonga replied:
Hi Ben using allOf checks that particular key but since in this refactor profileId is wrapped into its own key 'allOf' will keep failing updating check to 'anyOf' will make it pass since we have 2 keys holding proto data. The check for 'profileId' is being done elsewhere on other tests.

BenHenning commented on app/src/sharedTest/java/org/oppia/android/app/topic/TopicFragmentTest.kt:
```diff
@@ -835,24 +767,22 @@ class TopicFragmentTest {
-    return launch<TopicActivity>(
-      createTopicPlayStoryActivityIntent(internalProfileId, topicId, storyId)
-    ).also { testCoroutineDispatchers.runCurrent() }
+    return launch(createTopicPlayStoryActivityIntent(profileId, topicId, storyId))
```
It seems that the previous runCurrent call was removed--please re-add it.

KevinGitonga replied:
Re-added this thanks.

BenHenning commented on app/src/sharedTest/java/org/oppia/android/app/topic/TopicFragmentTest.kt:
It seems that there are extra changes in this file--please fix.

KevinGitonga replied:
Noted updated. Thanks.

BenHenning commented on app/src/sharedTest/java/org/oppia/android/app/settings/profile/ProfileEditFragmentTest.kt:
Looks like this file has extra changes, too--please fix.

KevinGitonga replied:
Updated thanks.

BenHenning commented on app/src/sharedTest/java/org/oppia/android/app/profileprogress/ProfileProgressFragmentTest.kt:
It looks like this file has extra changes, too--please fix.

KevinGitonga replied:
Thanks, couldn't exactly note the extra changes on this file please clarify.

BenHenning commented on app/src/sharedTest/java/org/oppia/android/app/settings/profile/ProfileEditActivityTest.kt:
It looks like this file has extra changes, too--please fix.

KevinGitonga replied:
Noted, updated.

BenHenning commented on app/src/sharedTest/java/org/oppia/android/app/profile/AdminPinActivityTest.kt:
```diff
@@ -168,20 +174,32 @@ class AdminPinActivityTest {
+      adminPinEnum = 0
+    ).extractCurrentUserProfileId()
+
+    assertThat(profileId).isEqualTo(ProfileId.getDefaultInstance())
```
Here & everywhere else in the PR: when we're testing that fragments/activities correctly include profile IDs, we really should never be checking against default instance since that's also a failing case if the proto is absent from the bundle.

Other tests don't need to change (since it can be valid to depend on a profile ID of 0), but the new tests that are specifically verifying profile ID presence ought to verify a non-zero profile ID.

KevinGitonga replied:
Noted.

BenHenning commented on app/src/sharedTest/java/org/oppia/android/app/ongoingtopiclist/OngoingTopicListActivityTest.kt:
```diff
@@ -154,11 +158,37 @@ class OngoingTopicListActivityTest {
+  }
+
+  @Test
+  fun testOngoingTopicListActivity_createFragmentWithProfileId_verifyProfileIdInBundle() {
```
Prefer creating a new test suite for the fragment to test this there, rather than testing it indirectly through the activity's test suite.

KevinGitonga replied:
Hi Ben, this Activity does not have test Fragment classes in the codebase, I think this is the same since the activity by contains the fragment by default once created.

BenHenning commented on app/src/sharedTest/java/org/oppia/android/app/home/HomeActivityTest.kt:
```diff
@@ -1012,16 +1040,21 @@ class HomeActivityTest {
+        assertThat(
+          activity.intent.extractCurrentUserProfileId()
+            .internalId
+        ).isEqualTo(profileId1.internalId)
```
Here & elsewhere in the PR: prefer using TurhProtoSubject's assertThat() and verifying the whole profile ID rather than its internal ID.

KevinGitonga replied:
Noted.

BenHenning commented on app/src/sharedTest/java/org/oppia/android/app/home/HomeActivityTest.kt:
```diff
@@ -238,14 +240,36 @@ class HomeActivityTest {
+  }
+
+  @Test
+  fun testHomeActivity_createFragmentWithProfileId_verifyProfileIdInBundle() {
```
Similar to a comment later in the PR: prefer creating a dedicated test suite for the fragment to verify this rather than verifying it through the activity's test suite.

KevinGitonga replied:
Hi Ben for this we have tests verifying the same in specific fragments for example this just verify's that SpotlightFragment which is instantiated while this activity is created has bundle. It is just like a reinforcement since the underlying Fragments also have similar tests to assert the same.

BenHenning commented on app/src/sharedTest/java/org/oppia/android/app/devoptions/MarkChaptersCompletedFragmentTest.kt:
It looks like this file has extra changes, too--please fix.

KevinGitonga replied:
Updated.

BenHenning commented on app/src/sharedTest/java/org/oppia/android/app/devoptions/MarkChaptersCompletedActivityTest.kt:
It looks like this file has extra changes, too--please fix.

KevinGitonga replied:
Updated.

BenHenning commented on app/src/sharedTest/java/org/oppia/android/app/completedstorylist/CompletedStoryListActivityTest.kt:
```diff
@@ -176,15 +179,43 @@ class CompletedStoryListActivityTest {
+  }
+
+  @Test
+  fun testCompletedStoryListActivity_createFragmentWithProfileId_verifyProfileIdInBundle() {
```
(See later comment--these should go in a dedicated test suite for the fragment).

KevinGitonga replied:
Noted.

BenHenning commented on utility/src/main/java/org/oppia/android/util/profile/CurrentUserProfileIdIntentDecorator.kt:
```diff
@@ -0,0 +1,48 @@
+    putProtoExtra(PROFILE_ID_INTENT_DECORATOR, profileId)
+  }
+
+  /** Returns the [ProfileId] packed in [this] intent or default instance if there is no profile ID bundled via [decorateWithUserProfileId]. */
```
Here & below: if this extends past 100 characters it needs to be line-wrapped.

KevinGitonga replied:
Updated.

BenHenning reviewed (dismissed):
Thanks @KevinGitonga! I checked over everything I could easily verify, but I had to skip 8 files that had extra changes. Please fix & address all comments, update to the latest develop, and verify CI is passing before sending back.

Only three old comments were kept open since I think they might not yet be fully resolved, but the other comment resolutions look good.

KevinGitonga committed:
Merge branch 'oppia:develop' into profile_id_provision_feature

KevinGitonga committed:
Integrate some changes from develop that were breaking this PR.

KevinGitonga committed:
Fix failing tests.

KevinGitonga committed:
Resolve issues per reviewer's comments.

KevinGitonga committed:
Refactor default instance to use 0.

KevinGitonga committed:
Refactor default instance to use 0.

KevinGitonga committed:
Add regex checks for ProfileId.

KevinGitonga committed:
Reformat activities to integrate with regex checks.

KevinGitonga committed:
Remove redundant Kdoc exemption.

KevinGitonga committed:
Remove redundant Kdoc exemption.

KevinGitonga committed:
Fix some failing tests after refactor.

KevinGitonga committed:
Fix some regex fail checks.

KevinGitonga committed:
Update test's.

KevinGitonga commented:
> Thanks @KevinGitonga! I checked over everything I could easily verify, but I had to skip 8 files that had extra changes. Please fix & address all comments, update to the latest develop, and verify CI is passing before sending back.
> 
> Only three old comments were kept open since I think they might not yet be fully resolved, but the other comment resolutions look good.

THanks Ben have updated this PR per the comments, Managed to resolve some failing tests on the regex checks what i am not able to resolve is one test because the output is not providing specific "Activity" which has regex issues check CI dump link below. REgex failing but the script is not pointing to any single point of failure for debbuging. PTAL and advice.
https://github.com/oppia/oppia-android/actions/runs/4616638018/jobs/8161953386?pr=4869

KevinGitonga committed:
Update test's.

BenHenning commented:
> > Thanks @KevinGitonga! I checked over everything I could easily verify, but I had to skip 8 files that had extra changes. Please fix & address all comments, update to the latest develop, and verify CI is passing before sending back.
> > Only three old comments were kept open since I think they might not yet be fully resolved, but the other comment resolutions look good.
> 
> THanks Ben have updated this PR per the comments, Managed to resolve some failing tests on the regex checks what i am not able to resolve is one test because the output is not providing specific "Activity" which has regex issues check CI dump link below. REgex failing but the script is not pointing to any single point of failure for debbuging. PTAL and advice. https://github.com/oppia/oppia-android/actions/runs/4616638018/jobs/8161953386?pr=4869

@KevinGitonga testFileNamePattern_activityInAppModule_fileNamePatternIsCorrect and testScreenNameTestPresence_activityTestWithScreenNameTest_screenNameTestIsPresent appear to be existing tests that may need to be updated since they are probably triggering the new regex pattern that was added.

BenHenning commented:
I plan to take a pass on this on Friday at the earliest, otherwise Monday.

BenHenning commented:
Update: I plan to look at this Tuesday/Wednesday.

KevinGitonga committed:
Merge branch 'develop' into profile_id_provision_feature

# Conflicts:
#	scripts/assets/file_content_validation_checks.textproto

KevinGitonga committed:
Revert "Merge branch 'develop' into profile_id_provision_feature"

This reverts commit f28d12ac006b87b957c23fddb1f83b764688c876, reversing
changes made to 3773844e8c87fc43d67039f875a377d89a4054ad.

KevinGitonga committed:
Revert "Revert "Merge branch 'develop' into profile_id_provision_feature""

This reverts commit 4b282c0f774459868eb20e69ab587a2de188330a.

KevinGitonga committed:
Revert "Merge branch 'develop' into profile_id_provision_feature"

This reverts commit f28d12ac006b87b957c23fddb1f83b764688c876, reversing
changes made to 3773844e8c87fc43d67039f875a377d89a4054ad.

KevinGitonga committed:
Revert "Revert "Merge branch 'develop' into profile_id_provision_feature""

This reverts commit 4b282c0f

KevinGitonga committed:
Merge remote-tracking branch 'origin/profile_id_provision_feature' into profile_id_provision_feature

KevinGitonga committed:
Merge branch 'develop' into profile_id_provision_feature

# Conflicts:
#	app/src/main/java/org/oppia/android/app/home/recentlyplayed/RecentlyPlayedFragmentPresenter.kt
#	scripts/assets/file_content_validation_checks.textproto

KevinGitonga committed:
Inline conflicting files during merge.

KevinGitonga committed:
Update some tests to prevent breaking tests after adding regex checks.

KevinGitonga committed:
Merge branch 'oppia:develop' into profile_id_provision_feature

KevinGitonga committed:
Merge branch 'develop' into profile_id_provision_feature

# Conflicts:
#	app/src/main/java/org/oppia/android/app/testing/ConceptCardFragmentTestActivity.kt
#	app/src/sharedTest/java/org/oppia/android/app/topic/conceptcard/ConceptCardFragmentTest.kt

KevinGitonga committed:
Fix pending failing test's.

KevinGitonga commented:
> Update: I plan to look at this Tuesday/Wednesday.

HI @BenHenning i have updated this to fix some of the pending failing tests, please take a pass. Thanks.

BenHenning commented:
Thanks @KevinGitonga. I'll need to take a look at this tomorrow or later in the week.

KevinGitonga committed:
Merge branch 'oppia:develop' into profile_id_provision_feature

BenHenning commented:
Since this overlaps so much with #4762, @KevinGitonga please assign back once:
- #4762 has been merged into develop
- This PR is then updated with the latest develop (to incorporate #4762 changes)
- CI is passing after updating to the latest develop that includes #4762 changes

KevinGitonga committed:
Merge branch 'oppia:develop' into profile_id_provision_feature

KevinGitonga committed:
Merge branch 'develop' into profile_id_provision_feature

# Conflicts:
#	app/src/main/java/org/oppia/android/app/administratorcontrols/appversion/AppVersionActivity.kt
#	app/src/main/java/org/oppia/android/app/administratorcontrols/learneranalytics/ProfileAndDeviceIdActivity.kt
#	app/src/main/java/org/oppia/android/app/completedstorylist/CompletedStoryListActivity.kt
#	app/src/main/java/org/oppia/android/app/completedstorylist/CompletedStoryListActivityPresenter.kt
#	app/src/main/java/org/oppia/android/app/completedstorylist/CompletedStoryListFragment.kt
#	app/src/main/java/org/oppia/android/app/completedstorylist/CompletedStoryListViewModel.kt
#	app/src/main/java/org/oppia/android/app/devoptions/DeveloperOptionsActivity.kt
#	app/src/main/java/org/oppia/android/app/devoptions/forcenetworktype/ForceNetworkTypeActivity.kt
#	app/src/main/java/org/oppia/android/app/devoptions/markchapterscompleted/MarkChaptersCompletedActivity.kt
#	app/src/main/java/org/oppia/android/app/devoptions/markchapterscompleted/testing/MarkChaptersCompletedTestActivity.kt
#	app/src/main/java/org/oppia/android/app/devoptions/markstoriescompleted/MarkStoriesCompletedActivity.kt
#	app/src/main/java/org/oppia/android/app/devoptions/markstoriescompleted/testing/MarkStoriesCompletedTestActivity.kt
#	app/src/main/java/org/oppia/android/app/devoptions/marktopicscompleted/MarkTopicsCompletedActivity.kt
#	app/src/main/java/org/oppia/android/app/devoptions/mathexpressionparser/MathExpressionParserActivity.kt
#	app/src/main/java/org/oppia/android/app/devoptions/vieweventlogs/ViewEventLogsActivity.kt
#	app/src/main/java/org/oppia/android/app/help/HelpActivity.kt
#	app/src/main/java/org/oppia/android/app/help/faq/faqsingle/FAQSingleActivity.kt
#	app/src/main/java/org/oppia/android/app/help/thirdparty/LicenseListActivity.kt
#	app/src/main/java/org/oppia/android/app/help/thirdparty/LicenseTextViewerActivity.kt
#	app/src/main/java/org/oppia/android/app/help/thirdparty/ThirdPartyDependencyListActivity.kt
#	app/src/main/java/org/oppia/android/app/mydownloads/MyDownloadsActivity.kt
#	app/src/main/java/org/oppia/android/app/ongoingtopiclist/OngoingTopicListActivity.kt
#	app/src/main/java/org/oppia/android/app/options/AppLanguageActivity.kt
#	app/src/main/java/org/oppia/android/app/options/OptionsActivity.kt
#	app/src/main/java/org/oppia/android/app/options/ReadingTextSizeActivity.kt
#	app/src/main/java/org/oppia/android/app/profile/AddProfileActivity.kt
#	app/src/main/java/org/oppia/android/app/profile/AdminAuthActivity.kt
#	app/src/main/java/org/oppia/android/app/profile/AdminPinActivity.kt
#	app/src/main/java/org/oppia/android/app/profile/PinPasswordActivity.kt
#	app/src/main/java/org/oppia/android/app/profileprogress/ProfilePictureActivity.kt
#	app/src/main/java/org/oppia/android/app/settings/profile/ProfileEditActivity.kt
#	app/src/main/java/org/oppia/android/app/settings/profile/ProfileListActivity.kt
#	app/src/main/java/org/oppia/android/app/settings/profile/ProfileRenameActivity.kt
#	app/src/main/java/org/oppia/android/app/settings/profile/ProfileResetPinActivity.kt
#	app/src/main/java/org/oppia/android/app/testing/AudioFragmentTestActivity.kt
#	app/src/main/java/org/oppia/android/app/testing/TopicRevisionTestActivity.kt
#	app/src/main/java/org/oppia/android/app/topic/TopicActivity.kt
#	app/src/main/java/org/oppia/android/app/topic/revisioncard/RevisionCardActivity.kt
#	app/src/main/java/org/oppia/android/app/walkthrough/WalkthroughActivity.kt
#	app/src/sharedTest/java/org/oppia/android/app/options/AppLanguageActivityTest.kt
#	app/src/sharedTest/java/org/oppia/android/app/options/AppLanguageFragmentTest.kt
#	app/src/sharedTest/java/org/oppia/android/app/options/AudioLanguageFragmentTest.kt
#	app/src/sharedTest/java/org/oppia/android/app/options/OptionsFragmentTest.kt
#	app/src/sharedTest/java/org/oppia/android/app/settings/profile/ProfileEditFragmentTest.kt
#	app/src/test/java/org/oppia/android/app/testing/options/OptionsFragmentTest.kt
#	scripts/assets/file_content_validation_checks.textproto
#	scripts/assets/kdoc_validity_exemptions.textproto

KevinGitonga committed:
Merge changes and clean PR after merge conflicts.

KevinGitonga committed:
Fix some failing tests.

KevinGitonga committed:
Merge branch 'oppia:develop' into profile_id_provision_feature

KevinGitonga committed:
Fix Kdoc issues.

KevinGitonga committed:
Merge remote-tracking branch 'origin/profile_id_provision_feature' into profile_id_provision_feature

KevinGitonga committed:
Merge branch 'oppia:develop' into profile_id_provision_feature

KevinGitonga committed:
Fix more failing tests.

KevinGitonga committed:
Merge branch 'develop' into profile_id_provision_feature

KevinGitonga committed:
Merge branch 'develop' into profile_id_provision_feature

KevinGitonga commented:
Hi @adhiamboperes updated this fully inline and merged with changes on develop which would be shelving this otherwise, PTAL and advice if it might be LGTM. Thanks.

KevinGitonga committed:
Update ExplorationActiveTimeControllerTest.kt to invoke startPlayinhNewExploration with ProfileId instead of internal id.

adhiamboperes commented on app/src/main/java/org/oppia/android/app/administratorcontrols/AdministratorControlsActivity.kt:
```diff
@@ -78,29 +81,30 @@ class AdministratorControlsActivity :
     }
-    val selectedProfileId = savedInstanceState?.getInt(SELECTED_PROFILE_ID_SAVED_KEY) ?: -1
+    profileId = savedInstanceState?.extractCurrentUserProfileId()
+      ?: ProfileId.newBuilder().apply { internalId = -1 }.build()
```
```suggestion
      ?: ProfileId.newBuilder().setInternalId(-1).build()
```

KevinGitonga replied:
Thanks, updated.

adhiamboperes commented on app/src/main/java/org/oppia/android/app/devoptions/forcenetworktype/ForceNetworkTypeActivity.kt:
```diff
@@ -28,9 +30,10 @@ class ForceNetworkTypeActivity : InjectableAutoLocalizedAppCompatActivity() {
   companion object {
     /** Returns [Intent] for [ForceNetworkTypeActivity]. */
-    fun createForceNetworkTypeActivityIntent(context: Context): Intent {
+    fun createForceNetworkTypeActivityIntent(context: Context, profileId: ProfileId): Intent {
       return Intent(context, ForceNetworkTypeActivity::class.java).apply {
         decorateWithScreenName(FORCE_NETWORK_TYPE_ACTIVITY)
+        decorateWithUserProfileId(profileId)
```
Is this intent profileID used somewhere? Since previously this intent did not need one?

KevinGitonga replied:
Just decorated as it may be required in future by child fragments and components for this particular Activity.

adhiamboperes commented on app/src/main/java/org/oppia/android/app/help/faq/faqsingle/FAQSingleActivity.kt:
```diff
@@ -32,11 +34,17 @@ class FAQSingleActivity : InjectableAutoLocalizedAppCompatActivity() {
     const val FAQ_SINGLE_ACTIVITY_ANSWER = "FAQSingleActivity.answer"
 
-    fun createFAQSingleActivityIntent(context: Context, question: String, answer: String): Intent {
+    fun createFAQSingleActivityIntent(
+      context: Context,
+      question: String,
+      answer: String,
+      profileId: ProfileId
+    ): Intent {
```
Similar question as above, is the aim just to decorate all the activities/fragments with the profileID even when not previously used in a specific instance?

KevinGitonga replied:
Yeah aim is to decorate entirely all activities as usages might occur in future per previous discussions.

adhiamboperes commented on app/src/main/java/org/oppia/android/app/profile/AdminPinActivityPresenter.kt:
```diff
@@ -120,7 +117,7 @@ class AdminPinActivityPresenter @Inject constructor(
                 activity.startActivity(
                   AdministratorControlsActivity.createAdministratorControlsActivityIntent(
-                    context, activity.intent.getIntExtra(ADMIN_PIN_PROFILE_ID_EXTRA_KEY, -1)
+                    context, activity.intent.extractCurrentUserProfileId()
```
Shouldn't you pass here the profileId variable from ln110?

KevinGitonga replied:
Yeah makes sense, Thanks.

adhiamboperes commented on app/src/main/java/org/oppia/android/app/profile/ProfileChooserFragmentPresenter.kt:
```diff
@@ -215,7 +216,7 @@ class ProfileChooserFragmentPresenter @Inject constructor(
           AdminPinActivity.createAdminPinActivityIntent(
             activity,
-            chooserViewModel.adminProfileId.internalId,
+            activity.intent.extractCurrentUserProfileId(),
```
For all the usages in this class, isn't it supposed to be `decorateWith` instead of extract? Also, could this be a global var in this class?

Are the profile IDs in `model.profile.id` different from the one from `activity.intent.extractCurrentUserProfileId()`

KevinGitonga replied:
Decorate packs the value in intent while extract unpacks the same.

adhiamboperes commented on app/src/main/java/org/oppia/android/app/settings/profile/ProfileEditDialogInterface.kt:
```diff
@@ -1,8 +1,10 @@
 
   /** Handles profile deletion in [ProfileEditFragment]. */
-  fun deleteProfileByInternalProfileId(internalProfileId: Int)
+  fun deleteProfileByInternalProfileId(profileId: ProfileId)
```
This either needs to remain unchanged, or the function should be renamed based on actual usage. Please review and update accordingly.

KevinGitonga replied:
Thanks, updated.

adhiamboperes reviewed (changes requested):
Hi @KevinGitonga, I did an initial pass and left some comments. PTAL!

KevinGitonga committed:
Merge branch 'develop' into profile_id_provision_feature

# Conflicts:
#	app/src/main/java/org/oppia/android/app/home/recentlyplayed/RecentlyPlayedFragmentPresenter.kt
#	app/src/sharedTest/java/org/oppia/android/app/administratorcontrols/learneranalytics/ProfileAndDeviceIdActivityTest.kt
#	app/src/sharedTest/java/org/oppia/android/app/profileprogress/ProfileProgressFragmentTest.kt

KevinGitonga committed:
Minor updates per reviewer's comments.

KevinGitonga commented:
Hi @adhiamboperes updated per the comments, might not fully resolve the static failures as i am using my mac which doesn't support bazel used in project last i checked. PTAL thanks

adhiamboperes committed:
Merge branch 'develop' into profile_id_provision_feature
